test/user-test.js

```diff
@@ -0,0 +1,183 @@
+            })
+            .then(res => {
+                expect(res).to.have.status(422);
+                done();
+            })
+            .catch(err => {
+                expect(err).to.not.be.null;
+                expect(err).to.have.status(422);
+                done();
+            });
+    });
+
+    it('Empty Email Return Error', (done) => {
+        request(apiAddress)
+            .post('/users')
+            .set('Content-Type', 'application/json')
+            .set('Accept', 'application/json')
+            .send({
+                email: '',
+                password: 'user1',
+                username: 'user1'
+            })
+            .then(res => {
+                expect(res).to.have.status(422);
+                done();
+            })
+            .catch(err => {
+                expect(err).to.not.be.null;
+                expect(err).to.have.status(422);
+                done();
+            });
+    });
+
+    it('Empty Password Return Error', (done) => {
+        request(apiAddress)
+            .post('/users')
+            .set('Content-Type', 'application/json')
+            .set('Accept', 'application/json')
+            .send({
+                email: 'user1@gmail.com',
+                password: '',
+                username: 'user1'
+            })
+            .then(res => {
+                expect(res).to.have.status(422);
+                done();
+            })
+            .catch(err => {
+                expect(err).to.not.be.null;
+                expect(err).to.have.status(422);
+                done();
+            });
+    });
+
+    it('Invalid Email Format Return Error', (done) => {
+        request(apiAddress)
+            .post('/users')
+            .set('Content-Type', 'application/json')
+            .set('Accept', 'application/json')
+            .send({
+                email: 'user1',
+                password: 'user1',
+                username: 'user1'
+            })
+            .then(res => {
+                expect(res).to.have.status(422);
+                done();
+            })
+            .catch(err => {
+                expect(err).to.not.be.null;
+                expect(err).to.have.status(422);
+                done();
+            });
+    });
+
+    it('Username Already Exist Return Error', (done) => {
+        request(apiAddress)
+            .post('/users')
+            .set('Content-Type', 'application/json')
+            .set('Accept', 'application/json')
+            .send({
+                email: 'userTest1@gmail.com',
+                password: 'userTest1',
+                username: 'userTest'
+            })
+            .then(res => {
+                expect(res).to.have.status(422);
+                done();
+            })
+            .catch(err => {
+                expect(err).to.not.be.null;
+                expect(err).to.have.status(422);
+                done();
+            });
+    });
+
+    it('Email Already Exist Return Error', (done) => {
```

Comment:
Replace 'Email Already Exist Return Error' to 'Returns error when Email is already exist'